components/membership/registration-impl/src/main/kotlin/net/corda/membership/impl/registration/dynamic/handler/mgm/StartRegistrationHandler.kt

```
@@ -234,27 +239,32 @@ internal class StartRegistrationHandler(
         command: StartRegistration
     ): net.corda.data.identity.HoldingIdentity? = state?.registeringMember
 
-    private class InvalidRegistrationRequestException(reason: String) : CordaRuntimeException(reason)
+    private class InvalidRegistrationRequestException(reason: String, val reasonForUser: String?) : CordaRuntimeException(reason)
 
-    private fun validateRegistrationRequest(condition: Boolean, errorMsg: () -> String) {
+    private fun validateRegistrationRequest(condition: Boolean, errorMsg: String, reasonForUser: String?) {
```

We can default the `reasonForUser` to be the `errorMsg`.

---

I considered that, but I decided it's better if we don't in order to force users of the API to think whether that is a sensible thing to do (as there are cases where sending the `errorMsg` can be problematic as it can leak sensitive information), instead of silently defaulting to it.